Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -433,13 +461,35 @@ public class ProgramBuilder {
                 }
             }
         }
-        
-        // Insert the slice into the currently mutated program
-        adopting(from: program) {
-            for instr in program {
-                if needs.contains(instr.index) {
-                    adopt(instr, keepTypes: true)
+
+        // This maps victim vars to host vars
+        var localVarMaps: [Variable: Variable] = [:]
+
+        // manual adoption
```

Comment:
I'm not really happy about this due to code duplication...
How about:
* You make a helper getter, `Program.lastVariable`, that gives you the last defined variable in a program (== the one with the highest number), and somewhere at the start of this function do `var placeholderVariable = program.lastVariable + 1` 
* Around line 400, you call `beginAdoption(from: program)`, and call `endAdoption()` at the end of this function
* When rewiring an input, you (1) replace the rewired variable with the next placeholder variable: `placeholderVariable++` and (2) manually create the placeholderVar -> rewiredVar mapping in `varMaps.last!` (feel free to create a new, private API for this)

Then here, you can just do `adopt(instr)` I think
WDYT? Would that work?